backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -92,6 +94,39 @@
                 .onItem().call(this::loadProperties);
     }
 
+    public Uni<EndpointType> getEndpointTypeById(String accountId, UUID endpointId) {
+        String query = "Select e.type from Endpoint e WHERE e.accountId = :accountId AND e.id = :endpointId";
+        return session.createQuery(query, EndpointType.class)
+                .setParameter("accountId", accountId)
+                .setParameter("endpointId", endpointId)
+                .getSingleResultOrNull();
+    }
+
+    public Uni<Endpoint> getOrCreateEmailSubscriptionEndpoint(String accountId, EmailSubscriptionProperties properties) {
+        return getEndpointsPerType(accountId, EndpointType.EMAIL_SUBSCRIPTION, true, null)
```

Comment:
I know email endpoints can't be disabled when this is merged, but that boolean add an extra condition to the HQL query for no reason 😃 

```suggestion
        return getEndpointsPerType(accountId, EndpointType.EMAIL_SUBSCRIPTION, false, null)
```

Reply:
Ah, you are right, we don't need to check for that.

Reply:
Actually, it has to be `null`, right?
```suggestion
        return getEndpointsPerType(accountId, EndpointType.EMAIL_SUBSCRIPTION, null, null)
```

Reply:
Good catch, I didn't pay enough attention to the method code 😃